src/structures/buffers/digest.rs

```diff
@@ -0,0 +1,66 @@
+use crate::response_code::{Error, Result, WrapperErrorKind};
+use crate::tss2_esys::TPM2B_DIGEST;
+use log::error;
+use std::convert::TryFrom;
+use std::ops::Deref;
+/// Struct holding a pcr value.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Digest {
+    value: Vec<u8>,
+}
+
+impl Deref for Digest {
+    type Target = Vec<u8>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.value
+    }
+}
```

Comment:
I do not know the documentation regarding Deref says the following:

"Implementing Deref for smart pointers makes accessing the data behind them convenient, which is why they implement Deref. On the other hand, the rules regarding Deref and DerefMut were designed specifically to accommodate smart pointers. Because of this, Deref should only be implemented for smart pointers to avoid confusion."

So I do not know if it makes sense to have it for the data types.

Reply:
Well I think `Auth` and `Data` are smart pointers and implementing `Deref` or `DerefMut` would make sense! On the same time, those structures will maybe be part of the zeroizing fest we were planning to do and maybe implement `Zeroize`? In which case, it would be fine not implementing `Deref` now and leaving it for later.